### DISCUSSION TITLE
Custom schedule

### DISCUSSION THREAD

**Discussion by ns-ggeorgiev** [renovatebot/renovate#23600] [Suggest an Idea]
### Tell us more.

We have periodic company events say "branch-cut," which occur every month. However, these events are somewhat flexible and can be skipped or delayed and then shifted to a later date. I would like to enable developers to schedule renovate activities around these events. For instance, they should be able to use a config like `schedule: "2 days after branch cut"` where the phrase "2 days after branch cut" refers to a centrally maintained configuration reflecting the dynamic changes related to the branch-cut event.

Is it currently possible to implement this functionality? If so, how? If not, can we have that?

**Comment by RahulGautamSingh** [comment]
Not possible currently. 

>  If not, can we have that?

Will need to wait for a comment from the maintainers 😶

**Comment by rarkins** [comment]
I think this would be best implemented inside the bot's `config.js` where you can apply custom logic to generate your config settings.

**Reply by ns-ggeorgiev** [reply]
My understanding is that config.js is only for global properties. Could you please elaborate how it can be used to provide custom schedule for different set of packages in multiple repos? Would I need to have all of these with custom schedule in that central config place?

**Reply by rarkins** [reply]
Yes, it's a global config.

Another possibility: have a centralized repo with presets, and then repos can extend whatever schedule preset they want. i.e. you make a commit any time it needs to change.

**Reply by ns-ggeorgiev** [reply]
That is exactly the goal for that feature. To have a preset that defines a custom schedule that individual repos can schedule on. How this can be done without a custom schedule feature?

Creating rules that match exact update targets will be challenging, or even impossible. Current set of matches do not allow for repo match. If I have one team that wants to schedule the same packages that another team does not want to, that will not only be hard, but literally impossible.

**Reply by rarkins** [reply]
I'm just suggesting using presets: https://docs.renovatebot.com/key-concepts/presets/

Then each team can choose which custom schedule (preset) to extend

**Reply by ns-ggeorgiev** [reply]
Based on my understanding how this works, say I have a preset `schedule all on branch cut`. When a team say I want to schedule my pip-compile, but I do not want to schedule, my docker, I need to create a second preset `schedule pip-compile on branch cut`, then another team say, I want to schedule my node but not this other thing, then I will have to create `schedule node on branch cut`.
And this will go like that for every sub-group any team would like to use the branch-cut schedule for.

I guess they can reset the schedule with exclude pattern in their repo. Is this what you suggesting? Or I am missing something.

**Reply by rarkins** [reply]
I don't really understand what you mean by branch cut, just suggesting ways you might be able to do it.

**Reply by ns-ggeorgiev** [reply]
sure, and I am trying to understand what you suggesting exactly, because you just pointing to existing features, you not exactly explaining how they can be used to provide the needed functionality.

`branch cut` is the custom event that does not aligned with the calendar that I would like to allow teams to schedule renovate activities on.

I would like to be able to define 
`
custom-schedule-events: {
     "branch-cut": ["08/1/2023",  "09/5/2023",  "10/3/2023",  "11/1/2023", "1/8/2024"]
}
`
and then when someone wants to have schedule that is aligned with this event they can say

`schedule: "on branch-cut"`
or
`schedule: "after 4pm on branch-cut"`

**Reply by rarkins** [reply]
We don't support such customization and I don't think we would want to unless there was wide demand for it.
